pilot/cmd/pilot-agent/main.go

```diff
@@ -452,6 +455,8 @@ func getDNSDomain(podNamespace, domain string) string {
 }
 
 func init() {
+	proxyCmd.PersistentFlags().StringVar(&platform, "platform", "Kubernetes",
```

Comment:
Strongly against adding new flags. 

And strongly against adding 'platform' just for this purpose. You need a setting for cred - not over-reach and add a poorly defined platform without any RFC.

Reply:
+1, can not approve any PR adding new flags. Everything should be mesh config or env var